pkg/base1/cockpit.js

```diff
@@ -1047,6 +1047,26 @@ function basic_scope(cockpit, jquery) {
      *    immediately. Wait until control is returned to the browser.
      */
 
+    var call_queue = [];
+    var call_timeout = null;
+
+    function call_drain() {
+        call_timeout = null;
+        var func;
+        for (;;) {
```

Comment:
This behavior is inconsistent: We have to control over the callback, so they could in theory add items to the queue. Also, adding (for whatever reason) something null/undefined to the array would result in almost impossible to debug behavior.
This will work if the function calls (and clears) exactly as many items on the stack as were on it when it started, i.e.: get length of call_queue first, then loop over those, call if not null, remove them all.